pkg/deviceplugin/api.go

```diff
@@ -96,6 +96,12 @@ type PostAllocator interface {
 	PostAllocate(*pluginapi.AllocateResponse) error
 }
 
+// PreferredAllocator is an optional interface implemented by device plugins.
+type PreferredAllocator interface {
+	// PreferredAllocation defines the list of devices preferred for allocating next.
+	PreferredAllocation(*pluginapi.PreferredAllocationRequest) (*pluginapi.PreferredAllocationResponse, error)
```

Comment:
This should be named `GetPreferredAllocation()` to mirror the spec for the RPC service.